src/Sulu/Bundle/AdminBundle/Resources/js/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+import React from 'react';
+import type {ChildrenArray, Element, ElementRef} from 'react';
+import {observer} from 'mobx-react';
+import {action, computed, observable} from 'mobx';
+import debounce from 'debounce';
+import Input from '../Input';
+import Popover from '../Popover';
+import Menu from '../Menu';
+import Suggestion from './Suggestion';
+import autoCompleteStyles from './autoComplete.scss';
+
+const LENS_ICON = 'search';
+const DEBOUNCE_TIME = 300;
+const POPOVER_HORIZONTAL_OFFSET = 5;
+const POPOVER_VERTICAL_OFFSET = -2;
+
+type Props = {
+    children: ChildrenArray<Element<typeof Suggestion>>,
+    /** The value inside the input field */
+    value: string,
+    placeholder?: string,
+    /** Shows the loading indicator when true */
+    loading?: boolean,
+    /** Called when a suggestion is set */
+    onChange: (value: string | number) => void,
+    /** Called with a debounce when text is entered inside the input */
+    onSearch: (query: string) => void,
+};
+
+@observer
+export default class AutoComplete extends React.PureComponent<Props> {
+    static defaultProps = {
+        value: '',
+    };
+
+    static Suggestion = Suggestion;
+
+    @observable inputRef: ElementRef<*>;
+
+    @observable inputValue: string = this.props.value;
+
+    dirtyValue: boolean = false;
+
+    componentWillReceiveProps(nextProps: Props) {
+        if (this.dirtyValue) {
+            this.dirtyValue = false;
+            this.setInputValue(nextProps.value);
+        }
+    }
+
+    componentWillUnmount() {
+        this.debouncedSearch.clear();
+    }
+
+    @computed get suggestionStyle(): Object {
+        const suggestionListMinWidth = (this.inputRef) ? this.inputRef.scrollWidth - POPOVER_HORIZONTAL_OFFSET * 2: 0;
+
+        return {
+            minWidth: Math.max(suggestionListMinWidth, 0),
+        };
+    }
+
+    createSuggestions(children: ChildrenArray<Element<typeof Suggestion>>) {
+        return React.Children.map(children, (child, index: number) => {
+            return (
+                <li
+                    style={this.suggestionStyle}
+                    className={autoCompleteStyles.suggestionItem}
+                >
+                    {
+                        React.cloneElement(child, {
+                            key: index,
+                            query: this.inputValue,
+                            onSelection: this.handleSuggestionSelection,
+                        })
+                    }
+                </li>
+            );
+        });
+    }
+
+    @action setInputValue(value: string) {
+        this.inputValue = value;
+    }
+
+    setInputRef = (inputRef: ElementRef<'label'>) => {
+        if (inputRef) {
+            this.inputRef = inputRef;
+        }
+    };
+
+    debouncedSearch = debounce((query: string) => {
+        this.props.onSearch(query);
+    }, DEBOUNCE_TIME);
+
+    handleSuggestionSelection = (value: string | number) => {
+        this.dirtyValue = true;
```

Comment:
Shouldn't `dirtyValue` be set to `false` once the suggestion has been selected? 🤔